src/Essentials/src/FilePicker/FilePicker.uwp.cs

```diff
@@ -42,13 +43,52 @@ async Task<IEnumerable<FileResult>> PlatformPickAsync(PickOptions options, bool
 
 			if (AppInfoUtils.IsPackagedApp)
 			{
-				foreach (var file in resultList)
-					StorageApplicationPermissions.FutureAccessList.Add(file);
+				AddToAndCleanUpFutureAccessList(resultList);
 			}
 
 			return resultList.Select(storageFile => new FileResult(storageFile));
 		}
 
+		void AddToAndCleanUpFutureAccessList(List<StorageFile> pickedFiles)
+		{
+			var fal = StorageApplicationPermissions.FutureAccessList;
+
+			try
+			{
+				// Check if (FutureAccessList current items + new picked files) is exceeding maximum items
+				if ((fal.Entries.Count + pickedFiles.Count) >= fal.MaximumItemsAllowed)
+				{
+					// We assume that the FutureAccessList items are saved in order, there is no metadata
+					// Therefore we start removing from the bottom of the list
+					var removeCount = Math.Min(fal.Entries.Count, pickedFiles.Count);
+					var falTokens = fal.Entries.TakeLast(removeCount).ToList();
+
+					foreach (AccessListEntry entry in falTokens)
+					{
+						fal.Remove(entry.Token);
+					}
+				}
+
+				// Check if the picked file count doesn't exceed the maximum, else take the last n number or picked files
+				var pickedFilesLimitedToMax = pickedFiles;
+				if (pickedFiles.Count > StorageApplicationPermissions.FutureAccessList.MaximumItemsAllowed)
+				{
+					pickedFilesLimitedToMax =
+						pickedFilesLimitedToMax.TakeLast((int)fal.MaximumItemsAllowed).ToList();
```

Comment:
Should we do the first X picked?

Reply:
Yeah it's a bit one or the other. I assumed; the last ones you picked are going to be the most recent ones. But if you get into this scenario you're most likely picking a bigger number of files in 1 operation so there is not really any "first" or "last" that matter then.